bindings/python/pymongocrypt/mongocrypt.py

```diff
@@ -794,21 +800,51 @@ def __init__(self, ctx, kms_providers, filter, provider, master_key,
             raise
 
 
+def _get_gcp_credentials():
+    """Get on-demand GCP credentials"""
+    metadata_host = "metadata.google.internal"
+    if os.getenv("GCE_METADATA_HOST"):
+        metadata_host = os.environ["GCE_METADATA_HOST"]
+    url = "http://%s/computeMetadata/v1/instance/service-accounts/default/token" % metadata_host
+
+    headers = {"Metadata-Flavor": "Google"}
+    response = requests.get(url, headers=headers)
+    if response.status_code != 200:
+        msg = "Unable to retrieve GCP credentials: expected StatusCode 200, got StatusCode: %s. Response body:\n%s" % (response.status_code, response.content)
+        raise ValueError(msg)
```

Comment:
ValueError is usually used to indicate an invalid argument was passed whereas here the problem is elsewhere. Perhaps MongoCryptError would be more appropriate here and below?

Reply:
Done